impeller/renderer/backend/metal/command_buffer_mtl.mm

```diff
@@ -171,6 +176,58 @@ static bool LogMTLCommandBufferErrorIfPresent(id<MTLCommandBuffer> buffer) {
   return true;
 }
 
+bool CommandBufferMTL::SubmitCommandsAsync(
+    std::shared_ptr<RenderPass> render_pass) {
+  TRACE_EVENT0("impeller", "CommandBufferMTL::SubmitCommandsAsync");
+  if (!IsValid() || !render_pass->IsValid()) {
+    return false;
+  }
+  auto context = context_.lock();
+  if (!context) {
+    return false;
+  }
+  [buffer_ enqueue];
+  auto buffer = buffer_;
+  buffer_ = nil;
+
+  auto worker_task_runner = ContextMTL::Cast(*context).GetWorkerTaskRunner();
+
+  auto mtl_render_pass = static_cast<RenderPassMTL*>(render_pass.get());
+
+  // Render command encoder creation has been observed to exceed the stack size
+  // limit for worker threads, and therefore is intentionally constructed on the
+  // raster thread.
+  auto render_command_encoder =
+      [buffer renderCommandEncoderWithDescriptor:mtl_render_pass->desc_];
+  if (!render_command_encoder) {
+    return false;
+  }
+
+  auto task = fml::MakeCopyable(
+      [render_pass, buffer, render_command_encoder, weak_context = context_]() {
+        auto context = weak_context.lock();
+        if (!context) {
+          return;
```

Comment:
I'm realizing this will not protect us from potentially encoding command in the background.

We need the gpu sync switch in here so we can fizzle if GPU access is disabled. That's available via `Shell::GetIsGpuDisabledSyncSwitch`. I think it would make sense to expose that on `ContextMTL`.

Reply:
which operations are forbidden in the background? Is just submitting the cmd buffer? I'm curious what stops us from doing that now on the single raster thread

Reply:
(that is, do we need to use the sync switch elsewhere too?)

Reply:
Committing encoding work is forbidden: https://developer.apple.com/documentation/metal/gpu_devices_and_work_submission/preparing_your_metal_app_to_run_in_the_background?language=objc

Before this change, the work would all be guarded by the rasterizer - but now, spawning new threads, it's harder to predict how scheduling will occur and the rasterizer may get to the end of the function it's calling but the task spawned to encode new work happens after.

Reply:
Specifically, `[buffer commit]` below is definitely illegal in the background, but I think `EncodeCommands` may make other calls that are probably not allowed.

At any rate, we can't trust that we're on the raster thread with the rasterizer having checked whether GPU access is allowed in this closure.